### PR TITLE
Introducing Symfony 2

### PR DESCRIPTION
**Pull request by DRvanR**
This is the first step towards the 5.x release, the introduction of the Symfony framework. The focus of this change is using Symfony over custom routing/dispatching/controller framework, reaching feature parity in a 100% backwards compatible way. 

Still to do:
- [x] update SAML2 lib to get better protection against Remote Entity Injection and Entity Expansion attacks
- [x] create accompanying PR on [OpenConext-deploy][1] (ensure cache dirs rotate with releases) => see https://github.com/OpenConext/OpenConext-deploy/pull/22

[1]: https://github.com/OpenConext/OpenConext-deploy

### CODE REVIEW

File: library/EngineBlock/Application/Bootstrapper.php [hunk 1]
```diff
@@ -174,7 +174,7 @@ protected function _bootstrapPhpSettings()
         $settings = $this->_application->getConfiguration()->phpSettings->toArray();
         $this->_setIniSettings($settings);
         // prevent any XXE attacks when processing XML
-        libxml_disable_entity_loader();
+//        libxml_disable_entity_loader();
```
Comment (rjkip) [inline library/EngineBlock/Application/Bootstrapper.php]: This has addressed elsewhere, I presume?

File: run-on-vm.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env sh
+
+cd ../OpenConext-deploy && vagrant ssh -c "cd /opt/openconext/OpenConext-engineblock/ && $*" && cd ../OpenConext-engineblock
```
Comment (rjkip) [inline run-on-vm.sh]: What if Vagrant command fails? You're left in `../OpenConext-deploy`. Protip:

```diff
- cd ../OpenConext-deploy && vagrant ssh -c "cd /opt/openconext/OpenConext-engineblock/ && $*" && cd ../OpenConext-engineblock
+(cd ../OpenConext-deploy && vagrant ssh -c "cd /opt/openconext/OpenConext-engineblock/ && $*";)
```

Parentheses execute the command(s) in a subshell.

File: src/OpenConext/EngineBlock/AuthenticationBundle/EventListener/RedirectToFeedbackpageExceptionListener.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+use EngineBlock_Corto_Module_Service_SingleSignOn_NoIdpsException;
+use EngineBlock_Corto_Module_Services_SessionLostException;
+use EngineBlock_Corto_ProxyServer_UnknownRemoteEntityException;
+use OpenConext\EngineBlock\CompatibilityBundle\Bridge\ErrorReporter;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class RedirectToFeedbackpageExceptionListener
```
Comment (rjkip) [inline src/OpenConext/EngineBlock/AuthenticationBundle/EventListener/RedirectToFeedbackpageExceptionListener.php]: Nit: `FeedbackPage`.

File: src/OpenConext/EngineBlock/AuthenticationBundle/Resources/config/routing.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+certificate:
+    resource: routing/certificate.yml
+
+feedback:
+    resource: routing/feedback.yml
+    prefix:   '/authentication/feedback'
+
+wayf:
+    resource: routing/wayf.yml
```
Comment (rjkip) [inline src/OpenConext/EngineBlock/AuthenticationBundle/Resources/config/routing.yml]: Separate routing files per concern, I likes it.

File: library/EngineBlock/Application/Bootstrapper.php (hunk 1 above)
Comment (DRvanR) [inline library/EngineBlock/Application/Bootstrapper.php]: Yes, in the SAML2 library, however that must still be updated (and this line removed), thanks for noti(ci)ng

File: run-on-vm.sh (hunk 2 above)
Comment (DRvanR) [inline run-on-vm.sh]: meh, quick hack is a quick hack :tongue: - will fix it!

File: theme/Gruntfile.js [hunk 5]
```diff
@@ -98,45 +82,29 @@ module.exports = function(grunt) {
...
-                    'rm ../www/profile/css',
-                    'rm ../www/profile/javascript'
-                ].join('&&') + ' || true'
-            },
-            material: {
-                command: [
-                    'rm ../www/profile/images',
-                    'rm ../www/profile/javascripts',
-                    'rm ../www/profile/stylesheets'
-                ].join('&&') + ' || true'
```
Comment (rjkip) [inline theme/Gruntfile.js]: Good catch. Could you remove the `grunt-shell` package?

File: theme/Gruntfile.js [hunk 6]
```diff
@@ -211,7 +178,7 @@ module.exports = function(grunt) {
 
             tasks.push('uglify:' + theme);
             tasks.push('add_comment:' + theme);
-            tasks.push('symlink:' + theme);
+            //tasks.push('symlink:' + theme);
```
Comment (rjkip) [inline theme/Gruntfile.js]: Nit.

File: app/AppCache.php [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+require_once __DIR__.'/AppKernel.php';
+
+use Symfony\Bundle\FrameworkBundle\HttpCache\HttpCache;
+
+class AppCache extends HttpCache
```
Comment (rjkip) [inline app/AppCache.php]: You could remove this I guess.

**Comment by rjkip** [comment]
Good amount of work you put into this. Awesome. Onwards and upwards from here! :balloon:

File: theme/Gruntfile.js (hunk 5 above)
Comment (DRvanR) [inline theme/Gruntfile.js]: Done in 036ea8d

**Comment by DRvanR** [comment]
@rjkip please check the last commits

File: src/OpenConext/EngineBlock/ApiBundle/Service/FeaturesService.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace OpenConext\EngineBlock\ApiBundle\Service;
+
+use OpenConext\EngineBlock\ApiBundle\Exception\InvalidArgumentException;
+
+class FeaturesService
```
Comment (rjkip) [inline src/OpenConext/EngineBlock/ApiBundle/Service/FeaturesService.php]: Ooh, fancy :blush:

File: src/OpenConext/EngineBlock/ApiBundle/Controller/ConnectionsController.php [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    ) {
+        $this->engineBlockApplicationSingleton = $engineBlockApplicationSingleton;
+        $this->authorizationChecker = $authorizationChecker;
+        $this->featuresService = $featuresService;
+    }
+
+    public function pushConnectionsAction(Request $request)
+    {
+        if (!$this->featuresService->metadataPushIsEnabled()) {
+            return new Response(404);
```
Comment (rjkip) [inline src/OpenConext/EngineBlock/ApiBundle/Controller/ConnectionsController.php]: You wot m8

(Response with 404 content rather than status code)

**Comment by rjkip** [comment]
Lookin' good.

**Comment by DRvanR** [comment]
@rjkip Fixed the response, can I :ship: it?

**Comment by rjkip** [comment]
:rocket::microscope:

**Pull request merged**